client/my-sites/site-settings/form-general.jsx

```diff
@@ -377,38 +402,46 @@ export class SiteSettingsFormGeneral extends Component {
 						) }
 					</FormSettingExplanation>
 				</FormLabel>
-				{ ! isNonAtomicJetpackSite && (
-					<>
-						<FormLabel className="site-settings__visibility-label is-private">
-							<FormRadio
-								name="blog_public"
-								value="-1"
-								checked={ -1 === blogPublic && ! wpcomComingSoon }
-								onChange={ () =>
-									this.handleVisibilityOptionChange( {
-										blog_public: -1,
-										wpcom_coming_soon: 0,
-									} )
-								}
-								disabled={ isRequestingSettings }
-								onClick={ eventTracker( 'Clicked Site Visibility Radio Button' ) }
-								label={ translate( 'Private' ) }
-							/>
-						</FormLabel>
-						<FormSettingExplanation isIndented>
-							{ hasLocalizedText(
-								'Your site is only visible to you and logged-in members you approve. Everyone else will see a log in screen.'
-							)
-								? translate(
-										'Your site is only visible to you and logged-in members you approve. Everyone else will see a log in screen.'
-								  )
-								: translate(
-										'Your site is only visible to you and logged-in members you approve.'
-								  ) }
-						</FormSettingExplanation>
-					</>
-				) }
-			</FormFieldset>
+			</Fragment>
+		);
+	}
+
+	privacyRadioComingSoon() {
+		const { fields, isRequestingSettings, eventTracker, translate } = this.props;
+		const wasComingSoon = parseInt( fields.was_wpcom_coming_soon, 10 ) === 1;
```

Comment:
Does `was_wpcom_coming_soon` mean that a site was formerly v1?

Reply:
Yup, this only supports v1. v2 settings are entirely ignored, the only thing enabling v2 does is switches on this functionality to disappear the UI option after a v1 user switches out of the setting. `was_wpcom_coming_soon` is just to store the intermediate state so it doesn't disappear until the user clicks save. Without `was_...` the UI element would disappear as soon as you change the option (but don't yet save.)